### PR TITLE
initial exercise to add new list items

### PR DESCRIPTION
**Pull request by MHolmes219**

### CODE REVIEW

File: todo_app/data/session_items.py [hunk 1]
```diff
@@ -53,6 +53,20 @@ def add_item(title):
 
     return item
 
+# TODO: Finish definition function
+def remove_item(item):
```
Comment (MHolmes219) [inline todo_app/data/session_items.py:57]: Started work on remove_items function, but not sure how to resolve the recursion issues.

File: todo_app/data/session_items.py [hunk 2]
```diff
@@ -54,6 +54,25 @@ def add_item(title):
...
+
+    Args:
+        item: The item to remove.
+    """
+    existing_items = get_items()
+
+    # Remove the item from the list
+    existing_items.remove(item)
+
+    updated_items = [item if item['id'] == existing_item['id'] else existing_item for existing_item in existing_items]
```
Comment (JoshAdamPowell) [inline todo_app/data/session_items.py]: Minor - These two lines seem to be doing very similar things! Either way should be correct but no need to have both 
so:
```
existing_items.remove(item)
session['items'] = existing_items
```

or
```
updated_items = [item for item in existing_items if item["id"] != int(id)]
session['items'] = updated_items
```

File: todo_app/templates/index.html [hunk 3]
```diff
@@ -7,12 +7,66 @@ <h1 class="display-4">To-Do App</h1>
...
     </div>
   </div>
-{% endblock %}
\ No newline at end of file
+  {% endif %}
+  <div class="row fixed-bottom py-2 bg-light border-top">
+    <div class="container">
+      <form id="new-item" method="post" action="/add-item">
+        <div class="input-group">
+          <input type="text" name="title" id="title" class="form-control" placeholder="Item title">
```
Comment (JoshAdamPowell) [inline todo_app/templates/index.html]: minor but might be nice to have a `required` attribute here so people don't submit empty tasks

**Review by JoshAdamPowell — APPROVED** [review]
This is really great work, got all the stretch goals in, some nice styling and even a separate `/view-item` page! I've left a couple of optional bits of feedback in the PR but more than happy for this to be merged as-is

File: todo_app/templates/index.html (hunk 3 above)
Comment (MHolmes219) [reply todo_app/templates/index.html]: Thanks, Josh. I've actioned your suggestions and also added a javascript function to auto focus the input field when the document is ready as I figured this is the primary action for the page, making it a bit more accessible.

**Pull request merged**